src/System.Net.Mail/src/System/Net/Mail/SmtpConnection.cs

```diff
@@ -641,7 +642,7 @@ private static void SendEHelloCallback(IAsyncResult result)
                             if ((e.StatusCode != SmtpStatusCode.CommandUnrecognized)
                                 && (e.StatusCode != SmtpStatusCode.CommandNotImplemented))
                             {
-                                throw e;
+                                throw;
```

Comment:
Right.  `throw;` can only be used inside of a `catch` block, and effectively continues the propagation of the exception that was caught (kind catch-and-release'ing a fish), so it retains its original info.  `throw e;` can be done anywhere, and overwrites the info.